pkg/core/blockchain.go

```diff
@@ -410,6 +413,64 @@ func (bc *Blockchain) init() error {
 	return bc.updateExtensibleWhitelist(bHeight)
 }
 
+// JumpToState is an atomic operation that changes Blockchain state to the one
+// specified by the state sync point p. All the data needed for the jump must be
+// collected by the state sync module.
+func (bc *Blockchain) JumpToState(p uint32) error {
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+
+	if !bc.stateSync.DataCollected() {
+		return fmt.Errorf("state sync module has wong state to perform state jump")
+	}
+
+	if p+1 >= uint32(len(bc.headerHashes)) {
+		return fmt.Errorf("invalid state sync point")
+	}
+
+	bc.log.Info("jumping to state sync point", zap.Uint32("state sync point", p))
+
+	block, err := bc.dao.GetBlock(bc.headerHashes[p])
+	if err != nil {
+		return fmt.Errorf("failed to get current block: %w", err)
+	}
+	err = bc.dao.StoreAsCurrentBlock(block, nil)
+	if err != nil {
+		return fmt.Errorf("failed to store current block: %w", err)
+	}
+	bc.topBlock.Store(block)
+	atomic.StoreUint32(&bc.blockHeight, p)
+	atomic.StoreUint32(&bc.persistedHeight, p)
```

Comment:
> How the node will recover from that?

We can go with a flag, but then we need to distinguish several stages of `JumpToState`, because it's impossible to recover from some of the stages. For example, imagine that node being killed during the process of `STTempStorage`->`STStorage` replacement. And imagine that some of the prefixes were replaced and some of them did not. Then we can't distinguish old ones from the new ones. The only possible solution I see is to drop the DB in this case.

At the same time we can easily handle recovering from other stages without dropping the DB.

So do you think that stages concept suits for these purposes?